server/index.js

```diff
@@ -0,0 +1,78 @@
+const express = require('express');
+const app = express();
+const cors = require('cors');
+const multer = require('multer');
+
+app.use(cors());
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+
+const upload = multer();
+
+const incrementObjCount = (key, obj) => {
+    obj[key] ? obj[key]++ : obj[key] = 1;
+};
+
+app.post('/', upload.single("file"), async (req, res) => {
+    if (req.file) {
+        const content = JSON.parse(req.file.buffer.toString());
+        const mediaUrlBase = "https://www.youtube.com/watch?v=";
+        const artistUrlBase = "https://www.youtube.com/channel/";
+
+        const mediaCount = {};
+        const artistCount = {};
+        const artistIdToTitle = {};
+
+        const parsedContent = {};
+        parsedContent['mediaPlays'] = content
+            .filter(vid => vid.header === "YouTube Music")
+            .map(song => {
+                const songId = song.titleUrl.slice(mediaUrlBase.length);
+                const artistIdName = song.subtitles.map(artist => ({id: artist.url.slice(artistUrlBase.length), name: artist.name}));
```

Comment:
I ran the server and tried it using Postman. It worked fine with the sample data, but when I used my real watch-history.json file, it crashed. 

I looked into it to see what caused this and apparently there's a YouTube Music entry without subtitle details that I did not include in the sample data file. I checked out the link, and apparently its not longer available. This leads me to think that we should not only filter only YouTube music entries, but YouTube Music entries with the **subtitles** property.

This is how the entry appears, but I removed the actual video id.
```json
{
  "header": "YouTube Music",
  "title": "Watched https://www.youtube.com/watch?v\someyoutubelink",
  "titleUrl": "https://www.youtube.com/watch?v\someyoutubelink",
  "time": "2021-01-17T02:09:22.235Z",
  "products": ["YouTube"],
  "activityControls": ["YouTube watch history"]
}
```

Reply:
![image](https://user-images.githubusercontent.com/22921369/165589277-2a2c9bd5-e4d4-4334-a7b0-40062cecd5f8.png)

I'm thinking that we could check for the YouTube Music Header and for the **subtitles** property as well. Maybe something like:

```suggestion
            .filter(vid => vid.header === "YouTube Music" && vid.subtitles)
            .map(song => {
                const songId = song.titleUrl.slice(mediaUrlBase.length);
                const artistIdName = song.subtitles.map(artist => ({id: artist.url.slice(artistUrlBase.length), name: artist.name}));
```